File: python/two-sum/two_sum.py
from pip import List


class TwoSum:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        past_values = {}

        for i in range(len(nums)):
            compliment = target-nums[i]
            if compliment in past_values:
                return [i, past_values[compliment]]
            past_values[nums[i]] = i
        return []

File: typescript/two-sum/src/two-sum.ts
function twoSum(nums: number[], target: number): number[] {
  const pastValues = new Map<number, number>();
  for (let i = 0; i < nums.length; i++) {
    const element = nums[i];
    const complement = target - element;
    if (pastValues.has(complement)) {
      return [i, pastValues.get(complement)!];
    }
    pastValues.set(element, i);
  }

  return [];
}
export default twoSum;
